src/models/passport/passport.ts

```diff
@@ -62,6 +62,12 @@ export class Passport {
     if (!this.has(path)) return;
     return get({ data: this.data, path })!;
   }
+
+  generic<T>(path: string): T | undefined {
```

Comment:
I'm planning another pass at this class and have just introduced this here to solve a few type problems.

I think this interface is essentially what I'd like a `Passport.get()` method to look like (but the internal logic would more closely map the current `get()` function.